operator/pkg/tasks/init/karmadaresource.go

```diff
@@ -138,8 +138,8 @@ func createCrds(crdsClient *crdsclient.Clientset, crdsPath string) error {
 }
 
 func patchCrds(crdsClient *crdsclient.Clientset, patchPath string, caBundle string) error {
-	for _, file := range util.ListFiles(patchPath) {
-		if file.IsDir() || path.Ext(file.Name()) != ".yaml" {
+	for _, file := range util.ListFileWithSuffix(patchPath, ".yaml") {
+		if file.IsDir() {
```

Comment:
may be this line is redundant, for your `ListFileWithSuffix` function has filtered `!info.IsDir()`